app/controllers/api/v1/graphql_controller.rb

```diff
@@ -149,7 +149,7 @@ def format_error_message(e)
       private
 
       def authenticate_graphql_user
-        params[:query].to_s.match(/^((query )|(mutation[^\{#]*{\s*(resetPassword|changePassword|resendConfirmation|userDisconnectLoginAccount)\())/).nil? ? authenticate_user! : authenticate_user
+        !!(params[:query].to_s.match(/^(query\s+Me\s*\{|(mutation[^{#]*{\s*(resetPassword|changePassword|resendConfirmation|userDisconnectLoginAccount)\())/)) ? authenticate_user : authenticate_user!
```

Comment:
I don't think the `!!` are needed, I ran your test, and it still works if you remove them.